src/int_utils.h

```diff
@@ -77,6 +76,19 @@ class BitWriter {
         offset += bits;
     }
 
+
+public:
+    BitWriter(unsigned char* output) : out(output) {}
+
+    template<int BITS, typename I>
+    inline void Write(I val) {
```

Comment:
Whether `inline` makes anything for templates?

Reply:
> * A hint for the compiler that inlining the function is desired.

I thought that C++ standard abandoned this meaning of the `inline` keyword, especially since C++17.

https://en.cppreference.com/w/cpp/language/inline

Of course, compilers could have different opinions :)